docs/src/types.md

```diff
@@ -58,3 +58,14 @@ Modules = [ManifoldsBase]
 Pages = ["numbers.jl"]
 Order = [:type, :function]
 ```
+
+## [Type Parameter](@id type-parameter)
+
+An [`AbstractManifold`](@ref) usually has a type parameter that determines e.g. its siye or
+more precisely the [`manifold_dimension`](@ref). The [`TypeParameter`](@ref ManifoldsBase.TypeParameter) offers the flexibility
+to have this parameter either as type parameter or a field
```

Comment:
```suggestion
Concrete [`AbstractManifold`](@ref)s usually correspond to families of manifolds that are parameterized by some numbers, for example determining their [`manifold_dimension`](@ref). Those numbers can either be stored in a field or as a type parameter of the structure. The [`TypeParameter`](@ref ManifoldsBase.TypeParameter) offers the flexibility
to have this parameter either as type parameter or a field.
```

Reply:
Nice, yours sounds better! We could take the chance to explain advantages of each approach?

As a parameter we can dispatch on it but it might lead to more (pre)compilations
As a field we avoid recompilations, but we can not dispatch.

Reply:
Good idea, I think your explanation is good. Except maybe that type parameter definitely leads to more compilation :wink: . I guess you could use exp on rotation matrices as an example.